### DISCUSSION TITLE
Trying to send two unrelated MIDI commands and they seem to disrupt each other

### DISCUSSION THREAD

**Discussion by tjwrona** [helgoboss/helgobox#1218] [Q&A]
I am trying to configure MIDI mapping from one program to another where I need to take MIDI input from the first program, remap it to a new value, and then send it to a hardware device. The issue is, ReaLearn seems to be getting kind of hung up when I try to send two commands within a relatively short period of time (within about a second or so)... I know my MIDI device can handle MIDI inputs much faster than every second, so I'm fairly certain the hangup is happening within ReaLearn and not with my MIDI device itself.

Here is an example of commands I am trying to send. First I will send a note (Note 40 on channel 2 for example)
![image](https://github.com/user-attachments/assets/92993feb-baa4-47c2-98b1-70301cd4d7a3)

This note gets remapped to a CC command to trigger a tone change on my MIDI device.

Then about a second later I will send another note (Note 40 on channel 3 for example)
![image](https://github.com/user-attachments/assets/329b1e94-cb78-41cb-9c5e-4803c41601a0)

This note gets remapped to a different CC command to trigger the activation of a stomp pedal on my MIDI device. (Note: For triggering a stomp pedal the actual value passed doesn't matter, it just needs to trigger that CC command with any value)

Both of these commands seem to work if I trigger them individually, but if I try to trigger one, then the other within a reasonably short period of time they seem to trip over each other and will trigger very inconsistently.

Any idea what is going on here?

**Comment by tjwrona** [comment]
I have tested in Reaper and I can send multiple MIDI commands to my hardware device very quickly and it will handle them all correctly when it comes from a normal MIDI track within Reaper.

This means the problem is definitely somewhere in my relearn configuration. I'm still trying to figure out exactly where it is going wrong.

I am watching through all of the tutorial videos now. Hopefully I will see something that seems like it may be related to the problem.

**Comment by helgoboss** [comment]
Two things:

1. Your first pattern is probably wrong. NOTE ON messages (and most other MIDI messages except sys-ex) have only 3 bytes, but you are sending 4. Remove the part in the brackets.
2. What's your input and output in ReaLearn?

**Comment by tjwrona** [comment]
I ended up getting it to work. Oddly the solution was to remove "04" from the target pattern for the first mapping and set the min/max of the target in the glue both to 4.

I don't know why this behaves differently, but doing this made things respond quick and snappy.

It's probably related to what you said. What do the brackets mean in the target "pattern" field? The brackets were automatically added

**Reply by helgoboss** [reply]
Brackets can contain dynamic parts of the message that will be replaced with the incoming value. See user guide.
